src/components/demos/write/Demo.js

```diff
@@ -55,91 +92,105 @@ export default function WriteDemo(props) {
-        `${binaryName}-${theme.arch}`,
-        usernamePasswordPairs
-      )
-      setApiOutput(output)
-      setAwaitingApi(false)
+      update({
+        writeDemo: {
+          ...demoState,
+          fetching: true,
+        },
+      })
+      // TODO allow execute to handle states e.g. isFetching isError...
+      const output = await execute(bin, usernamePasswordPairs)
+      update({
+        writeDemo: {
+          ...demoState,
+          fetching: false,
+          output,
+        },
+      })
     }
 
     if (usernamePasswordPairs.length > 0) {
       attemptLogin()
     }
-  }, [usernamePasswordPairs, execute, binaryName, theme])
+  }, [bin, demoState, execute, update, usernamePasswordPairs])
+
+  useEffect(() => {
+    update(initState)
+  }, [update])
 
   return (
     <>
       <Header {...props} showClose={true} />
       <Wrapper {...theme.wrapper}>
-        {successfulLogin(apiOutput) ? (
+        {successfulLogin(output) ? (
           <SecretDesktop icons={props.secretDesktop} />
         ) : (
           <Box {...demoState}>
             <Container
               styles={{ height: '100%', paddingTop: '150px' }}
               size={10}
             >
-              <LoginForm
-                demoState={demoState}
-                showSpinner={awaitingApi}
-                setUsernamePasswordPairs={setUsernamePasswordPairs}
-                apiOutput={apiOutput}
-              />
+              <LoginForm demoState={demoState} />
             </Container>
             <Help
               theme={theme}
               content={helpContent}
               showContentState={showHelp}
-              setShowContentState={setShowHelp}
             />
           </Box>
         )}
         {!isMorello
-          ? successfulLogin(apiOutput) && (
-              <ButtonSide action={switchToMorello} />
-            )
-          : (successfulLogin(apiOutput) || loginError(apiOutput)) && (
+          ? successfulLogin(output) && <ButtonSide action={switchToMorello} />
+          : (successfulLogin(output) || loginError(output)) && [
               <ButtonSide
+                key={'write-demo-side-btn'}
                 message={'Learn More'}
                 action={() => {
                   nav('/write-demo-explainer')
                 }}
-              />
-            )}
+              />,
+              <ConsoleButton
+                key={'write-demo-console-icon'}
+                update={update}
+                state={demoState}
+              />,
+            ]}
+        <Help theme={theme} content={helpContent} showContentState={showHelp} />
+        {showResponse && (
+          <Modal
+            type={'writeDemo'}
+            update={update}
+            show={true}
+            message={'Morellow authentication output'}
+            args={`${binaryName}-${theme.arch} ${usernamePasswordPairs.join(
+              ', '
+            )}`}
+            {...demoState}
+            {...props}
```

Comment:
Again that's a big state splat into props. Can we be a bit more refined about what we're passing?

Reply:
This is going to be merged same way as in demo1 to put all state related to -> `writeDemo`, props and writeDemo = demoState. Props are from `fixtures/demos.js` which is basically a demo answer file

Reply:
Merge both into one - demoState